lib/index.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.register = (server, options, next) => {
+
+  server.ext('onPreResponse', (request, reply) => {
+
+    const settings = request.route.settings.plugins.totalCount;
+    const credentials = request.auth.credentials;
+    const query = request.query;
+    const filter = request.query.filter || {};
+
+    if (settings && credentials && query.include && query.include.indexOf('total_count') !== -1 && request.response.source) {
+
+      Bluebird.resolve(settings.model)
+      .then((Model) => {
```

Comment:
Optional: is `Bluebird.resolve` actually necessary? IMO, only use it if we are actually allowing `settings.model` to be either a `Bookshelf.Model`, or a `Promise<Bookshelf.Model>`.